File: 2017/code/dictionaries/dictionaries.py
#!/usr/bin/python3
""" Python dictionaries """

def main():
    """ Main func """
    # dictionari alebo hashmap popripadne unordered map je asotiativny kontainer
    # kde kazdy kluc ma nejaku prisluchajucu hodnotu

    di1 = {}
    di2 = {1 : "one", 2 : "two", 3 : "three", 4 : "four", 5 : "five"}
    di3 = {6 : "six", 7 : "seven", 8 : "eight", 9 : "nine", 10 : "ten"}

    print("Type of di1: %s" % type(di1))
    print("Type of di2: %s" % type(di2))
    print("Type of di3: %s" % type(di3))

    # Do dictionary sa da pridavat
    di1["kluc"] = "hodnota"

    print(di1)

    # Z dictionary sa da mazat
    del di1["kluc"]

    print(di1)

    # dictionary sa daju updatovat
    di4 = di2.copy()
    di4.update(di3)
    print(di4)

    # skutocna sila je vo vyhladavani
    di5 = {"jedna" : 1, "dva" : 2, "tri" : 3, "styri" : 4}

    print(di4[di5["dva"]])

    # pripadne trochu dict comprehension
    di6 = {k: di4[v] for (k, v) in di5.items()}
    print(di6)

    # ako lahko sa daju prehodit kluce za hodnoty
    di7 = {v: k for (k, v) in di4.items()}
    print(di7)


if __name__ == '__main__':
    main()

File: 2017/code/dictionaries/dictionaries.java
import java.lang.System;

import java.lang.String;

import static java.util.stream.Collectors.*;
import java.util.stream.Stream;

import java.util.Hashtable;
import java.util.Dictionary;
import java.util.HashMap;

class Dictionaries
{
    public static < K,V > HashMap< V,K > swapKeysValues( HashMap<K,V> map ) {
        HashMap< V,K > rev = new HashMap< V, K >();
        for( HashMap.Entry< K,V > el : map.entrySet() )
        {
            rev.put(el.getValue(), el.getKey());
        }
        
        return rev;
    }

    public static void main(String[] args) {
        HashMap<String, String> di1 = new HashMap< String, String >();
        HashMap<Integer, String> di2 = new HashMap<Integer, String>(){ { put( 1, "one"); put( 2, "two" ); } };
        HashMap<Integer, String> di3 = new HashMap<Integer, String>(){ { put( 3, "three"); put( 4, "four" ); } };

        System.out.println( "Type of di1: " + di1.getClass().getName() );
        System.out.println( "Type of di2: " + di2.getClass().getName() );
        System.out.println( "Type of di3: " + di3.getClass().getName() );

        // Do dictionary sa da pridavat
        di1.put( "kluc", "hodnota" );

        System.out.println( di1 );

        // Z dictionary sa da mazat
        di1.remove("kluc");

        System.out.println( di1 );

        // dictionary sa daju updatovat
        HashMap<Integer, String> di4 = new HashMap< Integer, String >();
        di4.putAll( di2 );
        di4.putAll( di3 );

        System.out.println( di4 );

        // skutocna sila je vo vyhladavani
        HashMap<String, Integer> di5 = new HashMap< String, Integer >() {
            { put( "jedna", 1); put( "dva", 2); put( "tri", 3 ); put( "styri", 4); }
        };

        System.out.println( di4.get( di5.get( "dva" ) ) );

        // java nema dict comprehension

        // prehodenie prvkov
        System.out.println( swapKeysValues( di4 ) );
    }
} 
